lib/route.js

```diff
@@ -0,0 +1,367 @@
+'use strict'
+
+const FindMyWay = require('find-my-way')
+const proxyAddr = require('proxy-addr')
+const Context = require('./context')
+const { buildMiddie, onRunMiddlewares } = require('./middleware')
+const { hookRunner, hookIterator } = require('./hooks')
+const supportedMethods = ['DELETE', 'GET', 'HEAD', 'PATCH', 'POST', 'PUT', 'OPTIONS']
+const validation = require('./validation')
+const buildSchema = validation.build
+const { buildSchemaCompiler } = validation
+const { beforeHandlerWarning } = require('./warnings')
+
+const {
+  kRoutePrefix,
+  kLogLevel,
+  kHooks,
+  kSchemas,
+  kSchemaCompiler,
+  kContentTypeParser,
+  kReply,
+  kRequest,
+  kMiddlewares,
+  kGlobalHooks,
+  kDisableRequestLogging
+} = require('./symbols.js')
+
+function buildRouting (options) {
+  const router = FindMyWay(options.config)
+
+  const schemaCache = new Map()
+  schemaCache.put = schemaCache.set
+
+  let avvio
+  let fourOhFour
+  let trustProxy
+  let requestIdHeader
+  let querystringParser
+  let requestIdLogLabel
+  let logger
+  let hasLogger
+  let setupResponseListeners
+  let throwIfAlreadyStarted
+  let proxyFn
+  let modifyCoreObjects
+  let genReqId
+  let disableRequestLogging
```

Comment:
I have noticed that the source code almost use `var` instead of `let` in function scope. I wanna know why there use `let`?

Reply:
`var` is mainly used in hot code paths to optimize performance. This is setup code that only runs once so it's okay to use `let` since it's the more idiomatic way to declare variables in modern JavaScript.